### ISSUE TITLE
Marker plugin incorrect and inconsistent placement of HTML markers, imageLayer works perfectly fine with the same setup

### ISSUE THREAD

**Issue by Codename-404** [mistic100/Photo-Sphere-Viewer#1007]
### Describe the bug

I am trying to place markers dynamically on click. I have achieved this by converting the click position to sphere coords with your helper functions. 
This setup works perfectly fine with ImageLayer type, however with HTML type, it shows inconsistency in placement. The placement is incorrect. 

I have setup a demo project with minimal setup here: https://github.com/Codename-404/psv-html-bug-overview 

### Instructions: 
* Click on the screen to place markers. 
* Toggle comment on imageLayer and html to see how the same setup works for imageLayer but not for html.

### Online demo URL

https://github.com/Codename-404/psv-html-bug-overview

### Photo Sphere Viewer version

5.1.7

### OS & browser

MacOS & Chrome

**Comment by mistic100** [comment]
You don't need to manually compute the yaw and pitch, use `e.data.yaw` and `e.data.pitch`.

No idea why the position is wrong yet.

**Comment by Codename-404** [comment]
This is actually a demo setup of original project where I am using normal mouse event to implement dnd. Everything works fine, except the position of html.

One thing I noticed is, if you click on the same location multiple times, it will add new markers in a column, instead of overlaying on the same location.
I tested with only changing location of one marker instead of spawning multiple, then it results consistent position. It's still not correct, however consistent to the position clicked.
And it behaves inconsistently once you click on somewhere and it's overlapping with another html marker, then it positions itself inconsistently.

Hope these info helps. 
Thank you for looking into the issue ~
